### PR TITLE
Fixes ai being able to unglue things from people

### LINKED ISSUE #16011: AI Eye Can Unglue Things From People?
### Labels

[BUG][MINOR]

### Description

I clicked on something glued to someone and "Cool AI manages to unglue the message sign component from Barrel Punchbeef."

### Steps to reproduce

1.Find someone with a glue item
2.Click on it as an AI Eye
3.Become confused as an action bar appears
4.Unglue the item when it finishes

### Expected Behavior

AI eyes shouldn't be able to do this

### Additional Information & Screenshots

Reported by: Monoblaster
On server: Goonstation 1 Classic: Heisenbee
Active test merges: #15458, #15759, and #15833
Round log date: 2023-09-25-06-01
Reported on: 2023-09-25 06:25:42
Map: OSHAN

### PR DESCRIPTION
**Pull request by unfunnyperson**
## About the PR 
Stops the ai from being able to unglue items from things

## Why's this needed? 
Fixes #16011

### CODE REVIEW

File: code/datums/components/glued.dm [hunk 1]
```diff
@@ -76,7 +76,7 @@ TYPEINFO(/datum/component/glued)
 	qdel(src)
 
 /datum/component/glued/proc/on_attackhand(atom/movable/parent, mob/user)
-	if(user?.a_intent == INTENT_HELP)
+	if(user?.a_intent == INTENT_HELP && !isintangible(user))
```
Comment (ZeWaka) [inline code/datums/components/glued.dm:79]: This should be can_act / the other physical interaction check

**Review by ZeWaka — CHANGES_REQUESTED** [review]

**Review by ZeWaka — APPROVED** [review]

**Comment by ZeWaka** [comment]
> can_act and can_reach are unnecessary here, attackhand should not happen in the first place if that is the case

why are intangibles even getting through then

**Comment by pali6** [comment]
Because some stuff implements `ai_attack` as just a call to `AttackHand` (when the correct thing would be for both to call a third, new proc instead).

File: code/datums/components/glued.dm [hunk 2]
```diff
@@ -76,7 +76,7 @@ TYPEINFO(/datum/component/glued)
 	qdel(src)
 
 /datum/component/glued/proc/on_attackhand(atom/movable/parent, mob/user)
-	if(user?.a_intent == INTENT_HELP)
+	if(user?.a_intent == INTENT_HELP && !isintangible(user) && can_act(user) && can_reach(user, parent))
```
Comment (ZeWaka) [inline code/datums/components/glued.dm:79]:
```suggestion
	if(user?.a_intent == INTENT_HELP && !isintangible(user))
```
okay for now

**Review by ZeWaka — APPROVED** [review]

**Comment by ZeWaka** [comment]
so what would you recommend here @pali6

**Comment by pali6** [comment]
I don't really care either way. It's not like more checks hurt anything (most likely), they're just redundant

**Pull request merged**